src/base/compute/owl_computation_cpu_engine.ml

```diff
@@ -25,8 +25,9 @@ module Make_Nested
 
   (* core interface *)
 
+  (* ! Make sure the order of evaluation of init and eval is always the same. *)
   let eval_gen nodes =
-    Array.iter CG_Init._init_term nodes;
+    CG_Init._init_terms nodes;
     Array.iter CG_Eval._eval_term nodes
```

Comment:
A suggestion, given `CG_Init._init_terms nodes;` I think we can also add `CG_Eval._eval_terms nodes;`, for being consistent and better readability.

Reply:
That's right, I'll fix it.